client/app/components/template-explorer/template-explorer.component.js

```diff
@@ -0,0 +1,204 @@
+      templates: [],
+      limit: 20,
+      filterCount: 0,
+      filters: [],
+      templatesList: [],
+      selectedItemsList: [],
+      limitOptions: [5, 10, 20, 50, 100, 200, 500, 1000],
+      selectedItemsListCount: 0,
+      // Functions
+      resolveTemplates: resolveTemplates,
+     // listActionDisable: listActionDisable,
+      updatePagination: updatePagination,
+      // Config setup
+      actionConfig: getActionConfig(),
+      menuActions: getMenuActions(),
+      toolbarConfig: getToolbarConfig(),
+      listConfig: getListConfig(),
+      sortConfig: getSortConfig(),
+      offset: 0,
+      pollingInterval: 10000,
+    });
+
+    resolveTemplates(vm.limit, 0);
+    Polling.start('templateListPolling', pollTemplates, vm.pollingInterval);
+  }
+
+  function getActionConfig() {
+    return [
+    ];
+  }
+
+  function getListConfig() {
+    return {
+      useExpandingRows: false,
+      selectionMatchProp: 'id',
+      onCheckBoxChange: selectionChange,
+    };
+  }
+  function getSortConfig() {
+    return {
+      direction: 'asc',
+      field: 'name',
+      sortOptions: 'alpha',
+    };
+  }
+  function getToolbarConfig() {
+    var sortOrderFields = getOrderSortFields();
+    const sortConfig = {
+      fields: sortOrderFields,
+      onSortChange: sortChange,
+      isAscending: true,
+      currentField: sortOrderFields[0],
+    };
+
+    const filterConfig = {
+      fields: getOrderFilterFields(),
+      resultsCount: 0,
+      appliedFilters: [],
+      onFilterChange: filterChange,
+    };
+
+    return {
+      sortConfig: sortConfig,
+      filterConfig: filterConfig,
+      actionsConfig: {
+      },
+    };
+  }
+
+  function getOrderFilterFields() {
+    return [
+      ListView.createFilterField('name', __('Name'), __('Filter by Name'), 'text'),
+      ListView.createFilterField('type', __('Type'), __('Filter by Type'), 'text'),
+      ListView.createFilterField('draft', __('Draft'), __('Filter by Draft'), 'text'),
+    ];
+  }
+
+  function getOrderSortFields() {
+    return [
+      ListView.createSortField('name', __('Name'), 'alpha'),
+      ListView.createSortField('type', __('Type'), 'alpha'),
+    ];
+  }
+
+  function getMenuActions() {
+    const menuActions = [
+    ];
+
+    return menuActions;
+  }
+
+  function sortChange(sortId, direction) {
+    vm.sortConfig.field = sortId.id;
+    vm.sortConfig.direction = direction === true ? 'asc' : 'desc';
+    vm.sortConfig.sortOptions = sortId.sortType === 'alpha' ? 'ignore_case' : '';
+
+    resolveTemplates(vm.limit, 0);
+  }
+  function filterChange(filters) {
+    vm.filters = filters;
+    resolveTemplates(vm.limit, 0);
+  }
+  function resolveTemplates(limit, offset, refresh) {
+    if (!refresh) {
+      vm.loading = true;
+    }
+    var existingTemplates = (angular.isDefined(vm.templatesList) && refresh ? angular.copy(vm.templatesList) : []);
+    vm.offset = String(offset);
+
+    Templates.getMinimal(vm.filters).then(setResultTotals);
```

Comment:
We should add a `.catch` for each promise that can fail. You could use the `exception` module to easily add some log messages during failures.